libdisco/apis.go

```diff
@@ -123,21 +123,21 @@ var errNoProof = errors.New("Disco: no public key proof set in Config")
 
 func checkRequirements(isClient bool, config *Config) (err error) {
 	ht := config.HandshakePattern
-	if ht == Noise_NX || ht == Noise_KX || ht == Noise_XX || ht == Noise_IX {
+	if ht == NoiseNX || ht == NoiseKX || ht == NoiseXX || ht == NoiseIX {
```

Comment:
for now let's keep the underscore, this is what the documentation uses as well: https://www.discocrypto.com/#/protocol/Noise_N

Reply:
Hmm how about I upgrade both and make it consistent with Go idioms ? 
- https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps